proto/frequenz/api/reporting/v1/reporting.proto

```diff
@@ -0,0 +1,534 @@
+}
+
+// Response containing a single data sample for one microgrid's components.
+//
+// !!! note
+//     The microgrid's components are provided as single data samples that
+//     encapsulate metrics, bounds, errors, and operational states along with their
+//     associated timestamps. Each response message covers a single microgrid.
+//     If multiple microgrids are provided in the request, expect sequential messages
+//     in the stream.
+//
+// !!! example
+//     Example output structure:
+//     ```
+//     {
+//       microgrid_id: 1,
+//       components: [
+//         {
+//           component_id: 13,
+//           metric_samples: [
+//             /* all metrics belong to the same timestamp */
+//             /* sample is type MetricSampleVariant */
+//             { sampled_at: "2023-10-01T00:00:00Z", metric: "DC_VOLTAGE_V", sample: {...}, bounds: {...} },
+//             { sampled_at: "2023-10-01T00:00:00Z", metric: "DC_CURRENT_A", sample: {...}, bounds: {...} }
+//           ],
+//           state: {
+//             sampled_at: "2023-10-01T00:00:13.12Z",
+//             states: ["COMPONENT_STATE_CHARGING", "CABLE_CONNECTED"],
+//             warnings: []
+//             errors: [],
+//           ]
+//         },
+//         {
+//           component_id: 59,
+//           metric_samples: [
+//             /* all metrics belong to the same timestamp */
+//             /* sample is type MetricSampleVariant */
+//             { sampled_at: "2023-10-01T00:00:00Z", metric: "DC_VOLTAGE_V", sample: {...}, bounds: {...} },
+//             { sampled_at: "2023-10-01T00:00:00Z", metric: "DC_CURRENT_A", sample: {...}, bounds: {...} }
+//           ],
+//           state: {
+//             sampled_at: "2023-10-01T00:00:13.12Z",
+//             states: ["COMPONENT_STATE_ERROR"],
+//             warnings: []
+//             errors: ["COMPONENT_ERROR_CODE_BATTERY_RELAY_ERROR"],
+//           ]
+//         }
+//       ]
+//     }
+//     ```
+//
+message ReceiveMicrogridComponentsDataStreamResponse {
+  // Microgrid ID for which the components and samples are reported.
+  uint64 microgrid_id = 1;
+
+  // List of components within this microgrid, each with its associated data samples.
+  repeated frequenz.api.common.v1.microgrid.components.ComponentData components = 2;
+}
+
+// Message defining the request format for fetching aggregated historical metrics
+// for a list of microgrid components. This request allows to specify custom
+// aggregation formulas, along with general filtering and pagination options.
+//
+// !!! note
+//         The aggregation follows the passive sign convention. Bounds, operational
+//         state, or errors are not applicable to aggregated data.
+//
+message ListAggregatedMicrogridComponentsDataRequest {
+  // General filter criteria for querying aggregated microgrid components data.
+  message AggregationListFilter {
+    // Optional resampling options like resolution for the data, represented in seconds.
```

Comment:
I think resampling should be mandatory for aggregations unless we can be sure that the timestamps of components that are aggregated are in sync.

Reply:
These protobuf types are optional, so this documentation is correct.

But I agree with your point about samples being in sync in time. What we could specify is a default resampling period, in case clients do not specify it. I'd suggest 1s for now.

Reply:
Then we can add it to the docs now.

Reply:
I added a note about the default resampling period.

Reply:
Suggest: `Data will always be resampled to ensure consistency of the aggregation. If omitted, data will be resampled with a default of one second`.